pco/src/data_types/floats.rs

```diff
@@ -147,7 +150,121 @@ macro_rules! impl_float_number {
         l as Self
       }
     }
+  };
+}
+
+impl FloatLike for f16 {
```

Comment:
I wonder if separate `impl FloatLike for f16`, and by extension splitting of the `impl_float_like` macro, could've been avoided. I ran into a bunch of issues related to the use of `as` to convert to and from `f16` and well as simple assignments, e.g., `const ZERO: Self = 0.0;` and operations, e.g., `const MAX_FOR_SAMPLING: Self = Self::MAX * 0.5;`.

Reply:
Perhaps a better solutions would be to rely on `num_traits` which provides implementations for `abs`, `inv` (named `recip`), `round`, `exp2`, `from_f64`, `to_f64`, `max`, and `min`. What do you think?

Reply:
In the channel I said:

> Num traits might make sense, but would be a big change affecting more than just this. We'd have to experiment and see how it changes binary size. Let's not consider it for now

I still think this is right, and we should explore it in a separate branch later. This PR is already quite broad, and we shouldn't change too many things at once.

Reply:
Sorry, realized I'd put the comment the wrong place, so I moved it, only to see that you'd already replied. We'll keep `num_traits` out of this for now.